src/com/unbounce/treajure/java.clj

```diff
@@ -0,0 +1,20 @@
+(ns com.unbounce.treajure.java)
+
+(defprotocol ToClojure
+  (->clj [o]))
+
+(extend-protocol ToClojure
+  java.util.Map
+  (->clj [o] (let [entries (.entrySet o)]
```

Comment:
Yeah, I think that doesn't enforce that k is a String, is just a casting optimization (which will fail at runtime), there are legitimate uses of keys that are not `String`